polysh/rb_tree.py

```diff
@@ -40,20 +40,21 @@ def __init__(self, key = None, value = None, color = RED):
         self.value = value
         self.nonzero = 1
 
-    def __nonzero__(self):
-        return self.nonzero
+    def __bool__(self):
+        return bool(self.nonzero)
+
 
 class RBTree(object):
 
-    def __init__(self, cmpfn=cmp):
+    def __init__(self):
         self.sentinel = RBNode()
         self.sentinel.left = self.sentinel.right = self.sentinel
         self.sentinel.color = BLACK
         self.sentinel.nonzero = 0
         self.root = self.sentinel
         self.count = 0
         # changing the comparison function for an existing tree is dangerous!
-        self.__cmp = cmpfn
+        self.__cmp = lambda a, b: (a > b) - (a < b)
```

Comment:
We better change all occurrences of `self.cmp` to a simple comparison operator.  There is no point of keeping it if it is not going to be configurable.

Reply:
I made it configurable again then.

Reply:
I am sorry for confusing you.  I don't think making this configurable is a good idea either.  Python is an object oriented language.  The caller can define new types and define the way it is ordered.  Making this configurable independent from the type doesn't appear as a good design to me.  I assume it is not needed, anyway.  Also `libc` style `cmp` functions are not Pythonic anymore.

Reply:
Fair enough, I switched to pythonic comparisons and dropped the configurable comparison function.